utils/s2n_random.c

```diff
@@ -115,16 +116,24 @@ static __thread struct s2n_rand_state s2n_per_thread_rand_state = {
     .drbgs_initialized = false
 };
 
-static int s2n_rand_init_impl(void);
-static int s2n_rand_cleanup_impl(void);
-static int s2n_rand_urandom_impl(void *ptr, uint32_t size);
+int s2n_rand_init_impl(void);
+int s2n_rand_cleanup_impl(void);
+int s2n_rand_urandom_impl(void *ptr, uint32_t size);
```

Comment:
It seems really odd that you need access to these _impl methods. If this is absolutely necessary, you minimally need to give them better names.

Reply:
We discussed this offline. These fields are used to force the random implementation from rdrand to urandom. Even if we think that the CI might be running tests with urandom, it likely still makes sense to ensure that urandom is in use, since this test is specifically testing urandom.

However, we currently don't have a CI job that intentionally uses urandom when running all of the unit tests, so we should add that anyway. I opened an issue for this: https://github.com/aws/s2n-tls/issues/4366.

I did change the names of these functions, though. Let me know if they could still be more clear.